### PR TITLE
Define the grey-radiation advecting pulse test

### PR DESCRIPTION
**Pull request by chongchonghe**
### Description
Define the advecting pulse test (Krumholz2017) under grey-radiation approximation. The setup of the test is exactly the same as Sec. 4.1.6 of CASTRO III ([Zhang+2013](https://ui.adsabs.harvard.edu/abs/2013ApJS..204....7Z/abstract)). 

### Related issues
This PR corrected the formula of the v/c terms in the radiation flux update. The correct formula for the space-like component of the radiation four-force to order $\beta$ is

```math
-\boldsymbol{G} =-\rho \kappa_F \frac{\boldsymbol{F}_r}{c} + \rho \kappa_P\left(\frac{4 \pi B}{c} - E\right) \frac{\boldsymbol{v}}{c}+\rho \kappa_F \left (\frac{\boldsymbol{v}}{c} E + \frac{\boldsymbol{v} :\boldsymbol{P}_r}{c} \right).
```

instead of Eq. (8) of the Quokka paper, i.e.

```math
-\boldsymbol{G} =-\rho \kappa_F \frac{\boldsymbol{F}_r}{c} + \rho \kappa_P\left(\frac{4 \pi B}{c} \right) \frac{\boldsymbol{v}}{c}+\rho \kappa_F  \frac{\boldsymbol{v} :\boldsymbol{P}_r}{c} .
```

### Checklist
_Before this pull request can be reviewed, all of these tasks should be completed. Denote completed tasks with an `x` inside the square brackets `[ ]` in the Markdown source below:_
- [x] I have added a description (see above).
- [x] I have added a link to any related issues see (see above).
- [x] I have read the [Contributing Guide](https://github.com/quokka-astro/quokka/blob/development/CONTRIBUTING.md).
- [x] I have added tests for any new physics that this PR adds to the code.
- [x] I have tested this PR on my local computer and all tests pass.
- [x] I have manually triggered the GPU tests with the magic comment `/azp run`.
- [x] I have requested a reviewer for this PR.

### CODE REVIEW

**Comment by markkrumholz** [comment]
I'll review this PR tomorrow. Ben has asked me to take over reviewing these.

**Comment by chongchonghe** [comment]
> I'll review this PR tomorrow. Ben has asked me to take over reviewing these.

No worry. This PR relies on the [chongchong/new-Newton-iteration-basis](https://github.com/quokka-astro/quokka/tree/chongchong/new-Newton-iteration-basis) PR, so I'll wait till Ben approves that PR before I can merge this one.

File: src/RadhydroPulseGrey/test_radhydro_pulse_grey.cpp
Comment (markkrumholz) [inline src/RadhydroPulseGrey/test_radhydro_pulse_grey.cpp]: Is there a reason to write it this way instead of just setting all the elements to the values appropriate for v0 = 0? I ask just because when I read over this source code I was initially very confused why the initial condition setup routines for advecting and non-advecting were identical, and my confusion only resolved when I noticed that the two routines differed by exactly two characters.

File: src/RadhydroPulseGrey/test_radhydro_pulse_grey.cpp
Comment (markkrumholz) [inline src/RadhydroPulseGrey/test_radhydro_pulse_grey.cpp]: Why does periodic BC in the x direction not work?

File: src/radiation_system.hpp
Comment (markkrumholz) [inline src/radiation_system.hpp:1]: I guess this file contains the changes to the convergence structure in the earlier PR. I will wait until that is merged before reviewing this file, since those changes should then be merged in and this file will be much simpler.

**Review by markkrumholz — COMMENTED** [review]
I'll wait to review the changes to radiation_system.hpp until the PR changing the iteration method is merged in, but you can look at the comments on the other files now.

File: src/RadhydroPulseGrey/test_radhydro_pulse_grey.cpp
Comment (chongchonghe) [reply src/RadhydroPulseGrey/test_radhydro_pulse_grey.cpp]: Yes, I can do that. v0_nonadv is use only once and I can write v0 = 0 explicitly.

File: src/RadhydroPulseGrey/test_radhydro_pulse_grey.cpp
Comment (chongchonghe) [reply src/RadhydroPulseGrey/test_radhydro_pulse_grey.cpp]: I believe it was at some moment I found periodic B.C. does not work through testing, so I wrote this down. I just confirmed that both boundary conditions work equally well, so I'll stick with periodic BC .

**Comment by chongchonghe** [comment]
OK. I merged in the dev branch and resolved your comments. It's ready for your review now.

**Comment by chongchonghe** [comment]
Apart from adding the kappa_P - kappa_F term, what I have also done to the radiation_system.hpp is that I defined ComputeEddingtonTensor and call it from ComputeRadPressure and from computing the flux update. Pressure is simply T * erad.

**Review by markkrumholz — APPROVED** [review]

**Comment by markkrumholz** [comment]
Looks good to me. Ready to merge as soon as the GPU check finishes.

**Pull request merged**
